src/components/SendTransactionFlow.tsx

```diff
@@ -0,0 +1,350 @@
+import styled from "styled-components";
+import BigNumber from "bignumber.js";
+import { sendTxAction } from "ducks/sendTransaction";
+import { useDispatch } from "react-redux";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus } from "ducks/account";
+import { loadPrivateKey } from "helpers/keyManager";
+
+const El = styled.div``;
+
+const TempButtonEl = styled.button`
+  margin-bottom: 20px;
+`;
+
+const TempInputEl = styled.input`
+  margin-bottom: 20px;
+  min-width: 300px;
+`;
+
+const TempSelectInputEl = styled.select`
+  margin-bottom: 20px;
+  min-width: 300px;
+`;
+
+const TempAnchorEl = styled.a`
+  display: block;
+  margin-bottom: 20px;
+  cursor: pointer;
+  text-decoration: underline;
+`;
+
+// CREATE -> CONFIRM -> SUCCESS || ERROR
+const sendTxFlowEnum = {
+  CREATE: 0,
+  CONFIRM: 1,
+  SUCCESS: 2,
+  ERROR: 3,
+};
+
+interface FormData {
+  toAccountId: string;
+  amount: BigNumber;
+  fee: string;
+  memoType: string;
+  memoContent: string;
+}
+
+const initialFormData: FormData = {
+  toAccountId: "",
+  amount: new BigNumber(0),
+  fee: String(StellarSdk.BASE_FEE / 1e7),
+  memoType: StellarSdk.MemoNone,
+  memoContent: "",
+};
+
+export const SendTransactionFlow = () => {
+  const [currentStage, setCurrentStage] = useState(sendTxFlowEnum.CREATE);
+  const [formData, setFormData] = useState(initialFormData);
+
+  return (
+    <>
+      <div>
+        {currentStage === sendTxFlowEnum.CREATE && (
+          <div>
+            <CreateTransaction
+              onContinue={() => {
+                setCurrentStage(currentStage + 1);
+              }}
+              onInput={setFormData}
+              formData={formData}
+            />
+          </div>
+        )}
+      </div>
+      <div>
+        {currentStage === sendTxFlowEnum.CONFIRM && (
+          <El>
+            <ConfirmTransaction
+              onSuccessfulTx={() => {
+                setCurrentStage(sendTxFlowEnum.SUCCESS);
+              }}
+              onFailedTx={() => {
+                setCurrentStage(sendTxFlowEnum.ERROR);
+              }}
+              formData={formData}
+            />
+          </El>
+        )}
+      </div>
+      <div>
+        {currentStage === sendTxFlowEnum.SUCCESS && (
+          <El>
+            <SuccessfulTransaction
+              onRestartFlow={() => {
+                setFormData(initialFormData);
+                setCurrentStage(sendTxFlowEnum.CREATE);
+              }}
+            />
+          </El>
+        )}
+      </div>
+      <div>
+        {currentStage === sendTxFlowEnum.ERROR && (
+          <El>
+            <FailedTransaction
+              onEditTransaction={() => setCurrentStage(sendTxFlowEnum.CREATE)}
+            />
+          </El>
+        )}
+      </div>
+    </>
+  );
+};
+
+interface CreateProps {
+  onContinue: () => void;
+  onInput: (formData: FormData) => void;
+  formData: FormData;
+}
+
+const CreateTransaction = (props: CreateProps) => {
```

Comment:
We could move every component to its own file.

Reply:
The file is pretty long. I was trying to group them together under a "send flow" umbrella. To make it obvious it's used in that flow only. Thoughts on a "sendFlow" folder under components?